Use the New MatrixRTCSession MembershipManager

toger5 opened the pull request:
Fixes: https://github.com/element-hq/element-call/issues/3032
The is a developer mode option to switch back to the old/LegacyMembershipManager.

# ~~Blocked~~
~~This is currently blocked on an issue we observed on safari. Where safari is running into tight loop and as a result is slow/unusable.~~
 - [x] Add a fix that checks to not add multiple SendFirstEvents: https://github.com/matrix-org/matrix-js-sdk/pull/4726/commits/06545e810c3f5c9ccced58f966b0e49e9fd25d36
 - [x] Test the fix with safari
 
 **This has been solved in EC**

toger5 committed:
provide option to use the New MembershipManager

toger5 committed:
fix cryptoApi import change

toger5 committed:
add error screen

toger5 committed:
bump js-sdk

toger5 committed:
rename to `setUnrecoverableError` and remove onLeave call because that will be handled by an effect.

toger5 committed:
this was doing nothing (it is a fragment back when there was no deprecated `rtcSession.room`)

hughns commented on src/utils/errors.ts:
```diff
@@ -13,13 +13,16 @@ export enum ErrorCode {
   /** Calling is not supported, server misconfigured (JWT service missing, no MSC support ...)*/
   CONFIGURATION_ISSUE = "CONFIGURATION_ISSUE",
   NETWORK_CONNECTIVITY = "NETWORK_CONNECTIVITY",
+  RTC_SESSION_FAILIOUR = "RTC_SESSION_FAILIOUR",
```
Spelling:

```suggestion
  RTC_SESSION_FAILURE = "RTC_SESSION_FAILURE",
```

hughns commented on src/utils/errors.ts:
```diff
@@ -13,13 +13,16 @@ export enum ErrorCode {
   CONNECTION_LOST_ERROR = "CONNECTION_LOST_ERROR",
+  MEMBERSHIP_MANAGER_UNRECOVERABLE = "MEMBERSHIP_MANAGER_UNRECOVERABLE",
   UNKNOWN_ERROR = "UNKNOWN_ERROR",
+  // UNKNOWN_ERROR = "UNKNOWN_ERROR",
```
```suggestion
```

hughns commented on src/settings/submit-rageshake.ts:
```diff
@@ -9,10 +9,10 @@ import { type ComponentProps, useCallback, useEffect, useState } from "react";
   type MatrixClient,
   type MatrixEvent,
 } from "matrix-js-sdk/src/matrix";
+import { type CryptoApi } from "matrix-js-sdk/src/crypto-api";
```
I'm nor sure why these are in this PR? 🤷

toger5 replied:
Because we updated the js-sdk. And they recently deprecated non rust crypto and did some breaking changes.

toger5 committed:
rename to error

hughns commented on locales/en/app.json:
```diff
@@ -70,7 +70,8 @@
     "show_connection_stats": "Show connection statistics",
-    "show_non_member_tiles": "Show tiles for non-member media"
+    "show_non_member_tiles": "Show tiles for non-member media",
+    "use_NewMembershipmanager": "Use the new implementation of the call MembershipManager"
```
The name is inconsistent with other keys... but as it's a dev setting it's not important.

toger5 committed:
Update src/utils/errors.ts

Co-authored-by: Hugh Nimmo-Smith <hughns@users.noreply.github.com>

toger5 committed:
Update src/utils/errors.ts

Co-authored-by: Hugh Nimmo-Smith <hughns@users.noreply.github.com>

hughns commented on src/room/GroupCallView.tsx:
```diff
@@ -148,16 +172,17 @@ export const GroupCallView: FC<Props> = ({
     displayName,
     avatarUrl,
-    rtcSession.room,
+    rtcSession.room.roomId,
```
```suggestion
    room.roomId,
```

toger5 committed:
review

toger5 committed:
bump js-sdk

toger5 committed:
expose lk log level changing in `window`

toger5 committed:
bump js-sdk
 - always log "Missing own membership: force re-join"
 - also check insertions queue

toger5 committed:
Merge branch 'livekit' into toger5/test-with-new-membershipmanager

toger5 committed:
change lk log level to warn

hughns committed:
Bump js-sdk

hughns committed:
Bump js-sdk

hughns committed:
.

hughns committed:
Bump js-sdk

toger5 committed:
show user count based on meberships not users.

Signed-off-by: Timo K <toger5@hotmail.de>

toger5 committed:
Merge branch 'livekit' into toger5/test-with-new-membershipmanager

toger5 committed:
bump js-sdk

toger5 committed:
rename setting name

toger5 committed:
remove unused import

hughns commented on src/main.tsx:
```diff
@@ -24,6 +24,8 @@ import { App } from "./App";
 import { init as initRageshake } from "./settings/rageshake";
 import { Initializer } from "./initializer";
 
+window.setLKLogLevel = setLKLogLevel;
```
Can you stick this in a separate PR, please?

toger5 replied:
I was wondering the same. So yes!

toger5 committed:
js sdk bump

toger5 committed:
remove `window.setLKLogLevel`

toger5 committed:
bump js sdk with reverted incompatible change

toger5 committed:
bump js-sdk with one less merge

hughns reviewed (approved):
👍
